fgpv-vpgf/fgpv-vpgf
# fix(api): identify buffer size and layer queryable

## Description
Closes #2600 - buffer size can be set for each layer individually
Closes #2684 - idea 1 implemented
Starts on #2603 - moved `identifyMode` to from `Map` to `LayerGroup` and added `queryable` option to `ConfigLayer` which is linked with the UI query option in settings panel

## Testing
:eyes: 

To test identify mode: 
```js
`RZ.mapInstances[0].layers.identifyMode = 'none'` // turns off identify 
`RZ.mapInstances[0].layers.identifyMode = 'details'` // turns on identify 
```

To test field name in identify result (open the logged features array and check the contained data array):
```js
RZ.mapInstances[0].layers.identify.subscribe(event => {
    event.requests.forEach((request, index) =>
        request.features.then(features =>
            console.log(features)
        )
    );
});
```

To test layer `ConfigLayer` queryable:
```js
RZ.mapInstances[0].layers.getLayersById('powerplant100mw-liquids')[0].queryable = false // turns off query option
RZ.mapInstances[0].layers.getLayersById('powerplant100mw-liquids')[0].queryable = true // turns on option to query
// updates throughout the viewer
```

To test layer buffer size:
```js
RZ.mapInstances[0].layers.getLayersById('powerplant100mw-liquids')[0].identifyBuffer = 500 // why not
// click on map near a feature
```

## Documentation
In-line comments, JSDoc

## Checklist



Remember, it is a *muffin offence* to open a PR with any of the above checklist items incomplete.

Please keep the original issue up to date with the final solution, expected behaviour, and any additional notes for testers


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/2740)
